core/src/main/java/io/kestra/core/server/AbstractServiceLivenessTask.java

```diff
@@ -0,0 +1,139 @@
+import io.micronaut.core.annotation.Introspected;
+import jakarta.annotation.PostConstruct;
+import jakarta.annotation.PreDestroy;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Base class for scheduling a task that operate on Worker liveness.
+ */
+@Introspected
+@Slf4j
+public abstract class AbstractServiceLivenessTask implements Runnable, AutoCloseable {
+
+    private final String name;
+    protected final ServerConfig serverConfig;
+    private final AtomicBoolean isStopped = new AtomicBoolean(false);
+    private ScheduledExecutorService scheduledExecutorService;
+    private Instant lastScheduledExecution;
+
+    /**
+     * Creates a new {@link AbstractServiceLivenessTask} instance.
+     *
+     * @param name          the task name.
+     * @param configuration the liveness configuration.
+     */
+    protected AbstractServiceLivenessTask(final String name,
+                                          final ServerConfig configuration) {
+        this.name = Objects.requireNonNull(name, "name cannot be null");
+        this.serverConfig = Objects.requireNonNull(configuration, "serverConfig cannot be null");
+        this.lastScheduledExecution = Instant.now();
+    }
+
+    /**
+     * {@inheritDoc}
+     **/
+    @Override
+    public void run() {
+        final Instant now = Instant.now();
+        run(now);
+    }
+
+    @VisibleForTesting
+    public void run(final Instant now) {
+        try {
+            long elapsed = getElapsedMilliSinceLastSchedule(now);
+            long timeout = serverConfig.liveness().timeout().toMillis();
+            if (elapsed > timeout) {
+                // useful for debugging unexpected heartbeat timeout
+                log.warn("Thread starvation or clock leap detected (elapsed since previous schedule {}", elapsed);
+            }
+            onSchedule(now);
+        } catch (Exception e) {
+            log.error("Unexpected error while executing '{}'. Error: {}", now, e.getMessage());
+        } finally {
+            lastScheduledExecution = now;
+        }
+    }
+
+    protected Instant lastScheduledExecution() {
+        return lastScheduledExecution;
+    }
+
+    protected long getElapsedMilliSinceLastSchedule(final Instant now) {
+        return now.toEpochMilli() - lastScheduledExecution.toEpochMilli();
+    }
+
+    /**
+     * The callback method invoked on each schedule.
+     *
+     * @param now the time of the execution.
+     * @throws Exception when something goes wrong during the execution.
+     */
+    protected abstract void onSchedule(final Instant now) throws Exception;
+
+    /**
+     * Starts this task.
+     */
+    @PostConstruct
+    public synchronized void start() {
```

Comment:
```suggestion
    public void start() {
```

PostConstruct gives the needed concurrency safety